ethereum/blockcreation/src/main/java/tech/pegasys/pantheon/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -245,24 +245,25 @@ public boolean isCancelled() {
     return isCancelled.get();
   }
 
-  protected void throwIfStopped() throws CancellationException {
+  private void throwIfStopped() throws CancellationException {
     if (isCancelled.get()) {
       throw new CancellationException();
     }
   }
 
   /* Copied from BlockProcessor (with modifications). */
-  private boolean rewardBeneficiary(
+  boolean rewardBeneficiary(
       final MutableWorldState worldState,
       final ProcessableBlockHeader header,
       final List<BlockHeader> ommers,
-      final Wei blockReward) {
+      final Wei blockRewardSpec) {
 
     // TODO(tmm): Added to make this work, should come from blockProcessor.
     final int MAX_GENERATION = 6;
-    if (blockReward.isZero()) {
+    if (blockRewardSpec.isZero()) {
       return true;
     }
+    final Wei blockReward = blockRewardSpec.equals(Wei.NO_REWARD) ? Wei.ZERO : blockRewardSpec;
```

Comment:
Instead of having this special `Wei.NO_REWARD` value - what about just adding a flag like `forceUpdateBeneficiaryAccount(boolean shouldForceUpdate)` or something like that?  I think that would make the intentions behind this logic clearer.  Then you can return early on line 263 iff `blockRewards.isZero() && !forceUpdate`.

Reply:
We would need to add this flag to both the block creation and the block validation.  That would involve passing a flag into the ProtocolScheduleBuilder, and the question would be "is EIP 158 active?"

Reply:
I just pused an alternate approach where we expose EIP158 status in the ProtocolSpec, seems to get to both locations with a minimum of grief.